src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const areBothArgumentPassed = arguments.length === 2;
+    let total = this[0];
```

Comment:
The same for startIndex
```suggestion
    let total = arguments.length > 1 ? initialValue : this[0];
```